### ISSUE TITLE
Vermin reports that code is incompatible but it is

### ISSUE THREAD

**Issue by WhyNotHugo** [netromdk/vermin#230]
**Describe the bug**

Vermin reports that my codebase is not compatible with 3.8, but it is. Oddly, it points out:

    Minimum required versions: 2.3, 3.0

Which sounds like it should be fine to me.

**To Reproduce**

```python
if sys.version_info < (3, 8):  # noqa: UP036 # novermin
    errstr = "khal only supports python version 3.8+. Please Upgrade.\n"
    sys.stderr.write("#" * len(errstr) + '\n')
    sys.stderr.write(errstr)
    sys.stderr.write("#" * len(errstr) + '\n')
    sys.exit(1)
```

Vermin fails, indicating that this code doesn't run for the specified version:

```con
> vermin -t=3.8  --violations setup.py
Detecting python files..
Analyzing using 24 processes..
Minimum required versions: 2.3, 3.0
Target versions not met:   3.8
Note: Number of specified targets (1) doesn't match number of detected minimum versions (2).
```

**Expected behavior**

This should work just fine.

**Environment (please complete the following information):**

- Via pre-commit, v1.5.2.
- From Alpine packages, v1.5.2

**Additional context**

Full file is here: https://github.com/pimutils/khal/blob/f5ca8883f2e3d8a403d7a6d16fa4f552b6b69283/setup.py

**Comment by brenns10** [comment]
Yeah, I encountered the same issue - simple code which happens to be Python 2.x compatible will get flagged because I've only provided Python 3.x compatibility targets.

The issue is that there's basically a list of requirements, and a list of targets. The requirements includes an entry for Python 2.x and for 3.x. But you've only specified targets for 3.x. So when the code goes to check the requirements, it sees that the lengths of these two lists don't match, and fails. My pull request (#234) adjusts the logic to align the targets and requirements on their major version, so that the lists don't need to match in length.
